numpy/f2py/tests/test_crackfortran.py

```diff
@@ -88,6 +88,31 @@ def test_defaultPublic(self, tmp_path):
         assert 'private' not in mod['vars']['seta']['attrspec']
         assert 'public' in mod['vars']['seta']['attrspec']
 
+    def test_access_type(self, tmp_path):
+        f_path = tmp_path / "mod.f90"
+        with f_path.open('w') as ff:
+            ff.write(textwrap.dedent("""\
```

Comment:
Same nit as before, refactoring the test would be great as per #20517

Reply:
Sure, good point. Done.